### DISCUSSION TITLE
Can I insert a child element at a specified index or at the beginning of a ui.column?

### DISCUSSION THREAD

**Discussion by ItsCubeTime** [zauberzeug/nicegui#489] [Q&A] [answered]
Can I insert a child element at a specified index or at the beginning of a `ui.column` ?

```py
with ui.column as my_column:
    ui.label("there")

import time
time.sleep(1)
with my_column:
    ui.label("hello") # How would I make this appear before, "there"?
```

**Comment by falkoschindler** [comment, marked as answer]
Currently there is no official way to control the index of newly created elements in NiceGUI. But there are at least three workarounds:

1. In some cases it might be sufficient to create an empty container where the new element should appear:

	```py
	with ui.column() as column:
	    more = ui.column()
	    ui.label('first')

	with more:
	    ui.label('second')
    ```

2. If new elements should always appear above previous elements, you can set the flex direction of the container to "reverse":

	```py
	with ui.column().classes('reverse') as column:
	    ui.label('first')

	with column:
	    ui.label('second')
	```

	"reverse" is a Quasar class. The CSS property "flex-direction: column-reverse" works as well.

3. If you really need to move an element to a specific index, you can move the element in its container slot:

	```py
	with ui.column() as column:
	    ui.label('first')

	with column:
	    ui.label('second')
	    column.default_slot.children.insert(0, column.default_slot.children.pop())
	```

	In some cases you might need to call `column.update()` afterwards. But I'm not sure. Here it seems to work without.

**Reply by ItsCubeTime** [reply]
Adding to your 3rd example, would it be possible to insert also to the main page if Im not using a column at all?

Like:
```py
import nicegui
ui.label("there")
# Insert a new label "hello" above, "there"
ui.run()
```

Thank you for your very elaborate answer btw 😁It has been very helpful

**Reply by falkoschindler** [reply]
To move elements directly on the page (more precisely: the content of the auto-index client), you'd need to use the `get_client` function from the `globals` module:
```py
from nicegui import globals, ui

ui.label('there')
ui.label('above there')
globals.get_client().content.default_slot.children.insert(0, globals.get_client().content.default_slot.children.pop())

ui.run()
```
But be careful, NiceGUI's `globals` are mainly for internal use. Using a page function you can get the client as an argument:
```py
from nicegui import Client, ui

@ui.page('/')
def index_page(client: Client):
    ui.label('there')
    ui.label('above there')
    client.content.default_slot.children.insert(0, client.content.default_slot.children.pop())

ui.run()
```
